src/statsmodels/formula.jl

```diff
@@ -485,7 +485,16 @@ creating the model matrix.
         error("Could not construct model matrix. Resulting matrix has 0 columns.")
     end
 
-    ModelMatrix{T}(reduce(hcat, blocks), assign)
+    I = size(dfrm, 1)
+    J = mapreduce(x -> size(x, 2), +, blocks)
+    X = Array(eltype(T), I, J)
```

Comment:
Does it happen that `blocks[1]` is a sparse matrix though? Also, is the  code below, which fills the matrix column by column, reasonably efficient for sparse matrices?